rust/lance-file/src/v2/reader.rs

```diff
@@ -79,6 +91,9 @@ pub struct CachedFileMetadata {
     pub num_footer_bytes: u64,
     pub major_version: u16,
     pub minor_version: u16,
+
+    // The file statistics.
+    pub file_statistics: Arc<FileStatistics>,
```

Comment:
We could probably calculate these on demand.  I think 90% of the time users won't be accessing these and the cost to calculate them is trivial.

Reply:
changed to `calculate on demand`